### PR TITLE
Dependency refactor

### PR DESCRIPTION
**Pull request by alexfarrill**
- Slim down dependencies you have to add with yarn to none
- move types from dev dependencies to regular dependencies
- update readme to install mailing as a regular dependency
- some small updates to the e2e tests

e2e tests ran and verified that npx next build runs

### CODE REVIEW

**Review by psugihara — APPROVED** [review]
so sick!

**Comment by psugihara** [comment]
any conclusions on the best way to include `react` and `react-dom`?

File: packages/cli/README.md [hunk 1]
```diff
@@ -30,22 +30,22 @@ We’re longtime users of Action Mailer and wanted something similar for our typ
...
 
-1. Install mailing-core and the development server with yarn or npm:
+1. Install mailing with yarn or npm:
 
 yarn:
 
 ```
-yarn add mailing-core mjml mjml-react nodemailer &&\
-yarn add --dev mailing @types/mjml @types/mjml-react @types/nodemailer
+yarn add mailing
```
Comment (mrlubos) [inline packages/cli/README.md:38]: Sweet!

**Comment by mrlubos** [comment]
> any conclusions on the best way to include `react` and `react-dom`?

Can we mark all react dependencies as peer dependencies? That's what I've seen in other projects

**Comment by alexfarrill** [comment]
@mrlubos thanks for the peer dependency idea, I will check that out!

**Comment by alexfarrill** [comment]
@psugihara We already mark those react dependencies as pier dependencies in the CLI package. When I tested it in our branch it seemed to work though I’m not sure what changed from the other day when it didn’t. Give it a try if you have a second or we can look at it on Monday

**Comment by psugihara** [comment]
I get a module not found error with yarn 1.22.18 because it doesn't install peer deps. It [looks like some versions of npm do](https://stackoverflow.com/questions/35207380/how-to-install-npm-peer-dependencies-automatically), which might be what you're seeing.

```
> mkdir test && cd test && yarn init
> yarn add mailing
> yarn mailing
yarn run v1.22.18
$ /Users/petersugihara/test-standalone-install/node_modules/.bin/mailing
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'react-dom/server'
Require stack:
- /Users/petersugihara/test-standalone-install/node_modules/mjml-react/dist/src/index.js
- /Users/petersugihara/test-standalone-install/node_modules/mailing/dist/mailing.cjs.dev.js
- /Users/petersugihara/test-standalone-install/node_modules/mailing/dist/mailing.cjs.js
- /Users/petersugihara/test-standalone-install/node_modules/mailing/bin.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/petersugihara/test-standalone-install/node_modules/mjml-react/dist/src/index.js:8:38)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/petersugihara/test-standalone-install/node_modules/mjml-react/dist/src/index.js',
    '/Users/petersugihara/test-standalone-install/node_modules/mailing/dist/mailing.cjs.dev.js',
    '/Users/petersugihara/test-standalone-install/node_modules/mailing/dist/mailing.cjs.js',
    '/Users/petersugihara/test-standalone-install/node_modules/mailing/bin.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```


Not sure the best thing to do here. Perhaps it's catching this error and offering to install the deps? Otherwise, seems like we should call it out in the readme?

**Pull request merged**